src/components/dashboard/immunizationStatus/recommendationCard.tsx

```diff
@@ -43,7 +52,8 @@ export const RecommendationCard: FC<RecommendationCardProps> = ({
               <Badge
                 fontSize={'sm'}
                 colorScheme={getColorByStatus(
-                  configuration.forecastStatus.text
+                  configuration.forecastStatus.text as VaccinationStatus,
```

Comment:
@FuchsDominik The problem is that you are overriding the type checker here with `as VaccinationStatus`. Which means "please ignore everything you know about the type of the expression to the left and just trust me that it is indeed a `VaccinationStatus`". Without it you get:
```
TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'VaccinationStatus | undefined'.   Type 'string' is not assignable to type 'VaccinationStatus | undefined'.
```
Which is correct because `ImmunizationRecommendationRecommendation.forecastStatus.text` is of type `string`. This is a weaker type than `ImmunizationStatus`, so it cannot be inferred that it is safe to put this value into this argument.

Things like this is why we need a mapping layer between the FHIR DTOs and our application.

Reply:
Yes this was the issue I received when changing the method parameters, which was fixed when using the `as VaccinationStatus`. As we do not have a mapping layer yet, do you know a way to circumvent this error without using the `as` operator or should we go with this solution while the mapping layer is still under development?

Reply:
Moving this into a follow-up issue as we need to merge this to start working on the mapping layer.